src/Persistence/ItemPageLookupFactory.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Persistence;
+
+use ProfessionalWiki\WikibaseFacetedSearch\Application\Config;
+use Wikibase\Repo\WikibaseRepo;
+
+class ItemPageLookupFactory {
+
+	public function __construct(
+		private readonly Config $config
+	) {
+	}
+
+	public function newItemPageLookup(): ItemPageLookup {
+		if ( $this->config->linkTargetSitelinkSiteId !== null ) {
+			return new SiteLinkItemPageLookup(
+				WikibaseRepo::getStore()->newSiteLinkStore(),
+				$this->config->linkTargetSitelinkSiteId
+			);
+		}
+
+		return new NullItemPageLookup();
+	}
+}
```

Comment:
What motivated you to put this factory logic into a dedicated class? (I'm not objecting)

Reply:
I figured it would be slightly more testable than being directly in the extension factory (where we don't typically expect, and test, logic). But then again it's also slightly YAGNI since there are no other Lookups at the moment.